UI/SidePanel.cpp

```diff
@@ -3056,54 +3038,67 @@ void SidePanel::RefreshInPreRender() {
     s_system_connections.insert(GG::Connect(system->FleetsRemovedSignal,    &SidePanel::FleetsRemoved));
 }
 
+void SidePanel::RefreshSystemNames() {
+    TemporaryPtr<const System> system = GetSystem(s_system_id);
+    // if no system object, there is nothing to populate with.  early abort.
+    if (!system)
+        return;
+
+    // Repopulate the system with all of the names of known systems, if it is closed.
+    // If it is open do not change the system names because it runs in a seperate ModalEventPump
+    // from the main UI.
+    if (!m_system_name->Dropped()) {
+        m_system_name->Clear();
+
+        // populate droplist of system names
+        std::map<std::string, int> system_map; //alphabetize Systems here
+        for (ObjectMap::const_iterator<System> sys_it = Objects().const_begin<System>();
+             sys_it != Objects().const_end<System>(); ++sys_it)
+        {
+            if (!sys_it->Name().empty() || sys_it->ID() == s_system_id) // skip rows for systems that aren't known to this client, except the selected system
+                system_map.insert(std::make_pair(sys_it->Name(), sys_it->ID()));
+        }
+        std::vector<GG::DropDownList::Row*> rows;
+        rows.reserve(system_map.size());
+        for (std::map< std::string, int>::iterator sys_it = system_map.begin(); sys_it != system_map.end(); ++sys_it)
+        {
+            int sys_id = sys_it->second;
+            rows.push_back(new SystemRow(sys_id));
+        }
+        m_system_name->Insert(rows, false);
+
+        // select in the list the currently-selected system
```

Comment:
does inserting invalidate other row iterators? if no, can you select when adding instead of as a separate step?